src/app/accounts/settings.component.ts

```diff
@@ -269,17 +273,40 @@ export class SettingsComponent implements OnInit {
     }
 
     async saveCloseToTray() {
+        if (this.requireEnableTray) {
+            this.enableTray = true;
+        }
+
         await this.storageService.save(ElectronConstants.enableCloseToTrayKey, this.enableCloseToTray);
         this.callAnalytics('CloseToTray', this.enableCloseToTray);
     }
 
     async saveTray() {
+        if (this.requireEnableTray && !this.enableTray && (this.startToTray || this.enableCloseToTray)) {
+            const confirm = await this.platformUtilsService.showDialog(
+                this.i18nService.t('confirmTrayDesc'), this.i18nService.t('confirmTrayTitle'),
+                this.i18nService.t('yes'), this.i18nService.t('no'), 'warning');
+
+            if (confirm) {
+                this.startToTray = false;
+                this.enableCloseToTray = false;
+            } else {
+                this.enableTray = true;
+            }
+
+            return;
+        }
+
         await this.storageService.save(ElectronConstants.enableTrayKey, this.enableTray);
         this.callAnalytics('Tray', this.enableTray);
         this.messagingService.send(this.enableTray ? 'showTray' : 'removeTray');
     }
 
     async saveStartToTray() {
+        if (this.requireEnableTray) {
+            this.enableTray = true;
```

Comment:
Since this method isn't also persisting `this.enableTray` to the storage service, would we lose that setting if the user simply exited the app or negated that screen without saving tray settings?

Reply:
Whops, correct, we should save it.